### PR TITLE
Change to compose setup, restructured code, implementation of apis, dar…

### PR DESCRIPTION
**Pull request by hafizsaadiftikhar**
…k theme, currencies selection

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,12 +1,7 @@
 *.iml
-.gradle
+.gradle/*.*
+.idea/*.*
```
Comment (hank-milliken) [inline .gitignore:3]: There is now a lot of `.idea` files committed in this PR. This is probably not working

File: app/src/androidTest/java/com/codesses/priceconverter/ExampleInstrumentedTest.kt [hunk 2]
```diff
@@ -1,11 +1,13 @@
-package com.minimalisticapps.priceconverter
+package com.codesses.priceconverter
```
Comment (hank-milliken) [inline app/src/androidTest/java/com/codesses/priceconverter/ExampleInstrumentedTest.kt:1]: why this rename? it should be kept as `minimalisticapps`

File: app/src/main/java/com/codesses/priceconverter/common/utils/AppConstants.kt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package com.codesses.priceconverter.common.utils
+
+object AppConstants {
+    const val BIT_PAY_BASE_URL = "https://api.coingecko.com/"
```
Comment (hank-milliken) [inline app/src/main/java/com/codesses/priceconverter/common/utils/AppConstants.kt:4]: Those constants should be part of the "plugin" code. It should be easy to add new price sources

Comment (hank-milliken) [reply app/src/main/java/com/codesses/priceconverter/common/utils/AppConstants.kt:4]: See current implementation, every source of rates/prices has own class that shares interface

File: app/src/main/java/com/codesses/priceconverter/common/utils/PCSharedStorage.kt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class PCSharedStorage : SharedPrefHelper(PCSharedConfig.PREF_NAME) {
+
+    companion object {
+        private const val RECENT_SCREEN = "recent_screen"
+        private const val SLIDER_VISITED = "on_boarding_slider_visited"
+        private const val FILTERED_CITY = "filtered_city"
+        private const val EVENTS_CREDITS_LEFT = "events_credits_left"
+        private const val INVITATION_DAYS_LEFT = "invitation_days_left"
```
Comment (hank-milliken) [inline app/src/main/java/com/codesses/priceconverter/common/utils/PCSharedStorage.kt:11]: Why is this code here, how it is related to our app?

File: app/src/main/java/com/codesses/priceconverter/data/repo/PriceConverterRepositoryImpl.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import javax.inject.Inject
+
+class PriceConverterRepositoryImpl @Inject constructor(
+    private val api: ApiInterface
+) : PriceConverterRepository {
+    //
+//    override suspend fun getExchangeRates(): Flow<BitPayExchangeRatesResponse> = callbackFlow {
```
Comment (hank-milliken) [inline app/src/main/java/com/codesses/priceconverter/data/repo/PriceConverterRepositoryImpl.kt:18]: Probably will be deleted before merge, but just to keep track of it, lets not keep commented code, delete it instead.

File: app/src/androidTest/java/com/codesses/priceconverter/ExampleInstrumentedTest.kt (hunk 2 above)
Comment (hafizsaadiftikhar) [reply app/src/androidTest/java/com/codesses/priceconverter/ExampleInstrumentedTest.kt:1]: It's updated

File: app/src/main/java/com/codesses/priceconverter/di/AppModule.kt [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    @Singleton
+    fun provideHttpLogging(): HttpLoggingInterceptor {
+        return HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY)
+    }
+
+    @Provides
+    @Singleton
+    fun provideHttpClient(httpLoggingInterceptor: HttpLoggingInterceptor): OkHttpClient {
+        return try {
+            // Create a trust manager that does not validate certificate chains
```
[the remark this answers is not in the document]
Comment (hafizsaadiftikhar) [reply app/src/main/java/com/codesses/priceconverter/di/AppModule.kt]: It's for testing purpose

File: app/src/main/java/com/codesses/priceconverter/common/utils/PCSharedStorage.kt (hunk 4 above)
Comment (hafizsaadiftikhar) [reply app/src/main/java/com/codesses/priceconverter/common/utils/PCSharedStorage.kt]: It's for testing purposes. It will be resolved

**Comment by hafizsaadiftikhar** [comment]
Issues will be resolved in the next commit that is resolvable

**Pull request closed without merging**